src/OpenSearch.Net/Connection/HttpConnection.cs

```diff
@@ -120,7 +121,9 @@ public virtual TResponse Request<TResponse>(RequestData requestData)
 				if (responseMessage.Content != null)
 				{
 					receive = DiagnosticSource.Diagnose(DiagnosticSources.HttpConnection.ReceiveBody, requestData, statusCode);
-					responseStream = responseMessage.Content.ReadAsStreamAsync().GetAwaiter().GetResult();
+                    var task = Task.Run(() => responseMessage.Content.ReadAsStreamAsync());
```

Comment:
have you considered just changing it to responseMessage.Content.ReadAsStreamAsync().ConfigureAwait(false)?